### DISCUSSION TITLE
Howto make call/voice feature as close to GPT-4o voice advanced mode?

### DISCUSSION THREAD

**Discussion by stevenbaert** [open-webui/open-webui#4574] [Q&A]
Hi, 

I’m looking for advice on making the open-webui's call/voice mode as fast and interactive as possible. As close to advanced mode in GPT-4o voice as possible. 

**Here’s what I’m considering:**

1. **Speed:** I’m thinking of using Groq for faster response times. Ideally, I want the voice to start reading or outputting (streaming) text as soon as Groq generates it. Is this feasible?

2. **Voice Output:** Since OpenAI’s API might not be ideal for this, I’m considering local options. Would Coqui be a good choice for natural-sounding local voice synthesis? I’m also considering integrating local Whisper for quick and accurate speech recognition.

3. **Interactivity:** I’ve heard there’s work being done on making the voice output interruptible. Does anyone have updates on this? I’d love to be able to interrupt the assistant mid-task to issue new commands.

**Goal:** My goal is to use open-webui as a 24/7 assistant for managing tasks like emails, calendar events, and files. As it stands, it’s too slow for this purpose.

I’d appreciate any thoughts or suggestions on improving this setup!

Thanks!

S.

**Comment by winchanter85** [comment]
this is my current dream. Advanced voice mode helps a lot with language learning etc., but lots of people got stuck by the official app. GPT-4o Advanced Voiced Mode
kind of exclusive on chatgpt app for now according to this: https://help.openai.com/en/articles/9617425-advanced-voice-mode-faq

**Reply by stevenbaert** [reply]
Great that it is your dream and yes I know the details of advanced mode, but any plans or ideas of approach for Open-Webui to realize it?

**Comment by Pl8tinium** [comment]
i would love to see some kind of reconstruction of this feature. This is definetly something that is going to be essential in future LLM UI's

**Comment by i-iooi-i** [comment]
The current voice response is too slow, I use Azure Speech Service, after the voice is sent, it is parsed by the server, and finally the result is returned, which is too long. When will I get a fast response speed, I am really looking forward to it.
